### PR TITLE
US041 Authorisation System

### PR DESCRIPTION
**Pull request by Brigid481**

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -35,9 +36,9 @@ app.listen(3000, () => {
...
 });
 
-app.get('/openJobRoles', getAllJobRoles);
-app.get('/openJobRoles/:id', getJobRoleById);
-app.get('/homepage', getHomepage);
+app.get('/openJobRoles', allowRoles([UserRole.Admin, UserRole.User]), getAllJobRoles);
+app.get('/homepage', allowRoles([UserRole.Admin, UserRole.User]), getHomepage);
+app.get('/openJobRoles/:id', allowRoles([UserRole.Admin, UserRole.User]), getJobRoleById);
 
 app.get('/', (req: express.Request, res: express.Response) => {
```
Comment (gerardmclean2) [inline src/app.ts]: This probably isn't needed anymore. Was more of a placeholder for while the auth system wasn't in place

File: src/models/JwtToken.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export enum UserRole {
+    Admin = 1,
+    User = 2,
+
```
Comment (gerardmclean2) [inline src/models/JwtToken.ts]: No need for whitespace here

File: src/models/JwtToken.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export enum UserRole {
+    Admin = 1,
+    User = 2,
```
Comment (gerardmclean2) [inline src/models/JwtToken.ts]: Is the comma after the 2 necessary?

File: src/app.ts (hunk 1 above)
Comment (RaggedyRach) [reply src/app.ts]: left it in, just calling login. see what you think.

File: src/models/JwtToken.ts (hunk 3 above)
Comment (RaggedyRach) [reply src/models/JwtToken.ts]: removed

**Review by gerardmclean2 — APPROVED** [review]

**Review by boydstepkainos — APPROVED** [review]

**Pull request merged**